### ISSUE TITLE
[Bug] "cx_Freeze: Python error in main script" if compiler halts during startup

### ISSUE THREAD

**Issue by GitMensch** [OpenCobolIDE/OpenCobolIDE#163]
Because of an error with the compiler (which hasn't to do ANYTHING with OCIDE) the following message occurs

```
---------------------------
cx_Freeze: Python error in main script
---------------------------
Traceback (most recent call last):
  File "C:\Dev\OpenCobolIDE\open_cobol_ide\memoize.py", line 39, in __call__
KeyError: (<class 'open_cobol_ide.compilers.GnuCobolCompiler'>, 'C:\\Program Files (x86)\\OpenCobolIDE\\GnuCOBOL\\bin\\cobc')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\cx_Freeze\initscripts\Console.py", line 27, in <module>
  File "OpenCobolIDE", line 9, in <module>
  File "C:\Dev\OpenCobolIDE\open_cobol_ide\main.py", line 12, in main
  File "C:\Dev\OpenCobolIDE\open_cobol_ide\app.py", line 73, in __init__
  File "C:\Dev\OpenCobolIDE\open_cobol_ide\compilers.py", line 78, in check_compiler
  File "C:\Dev\OpenCobolIDE\open_cobol_ide\compilers.py", line 225, in is_working
  File "C:\Dev\OpenCobolIDE\open_cobol_ide\memoize.py", line 41, in __call__
  File "C:\Dev\OpenCobolIDE\open_cobol_ide\compilers.py", line 204, in check_compiler
  File "c:\python\32-bit\3.4\lib\encodings\cp1252.py", line 15, in decode
UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 146: character maps to <undefined>

---------------------------
OK   
---------------------------
```

After fixing this problem I found a very simple solution to reproduce the issue (using a working clean installation of 4.6.2 rc (signed) and a command prompt opened as administrator):

```cmd
cd %ProgramFiles(x86)%\OpenCobolIDE\GnuCOBOL\bin
move gcc.exe gcc_.exe
echo . > gcc.exe
..\..\OpenCobolIDE.exe
move /Y gcc_.exe gcc.exe
```

* close windows error message
* see error message from above

## System information

- OpenCobolIDE: 4.6.2
- Python: 3.4.3 (32bits)
- Qt: 5.4.1
- PyQt: 5.4.1
- pyqode.core: 2.7.0.dev2
- pyqode.cobol: 2.6.1
- pyqode.qt: 2.6.0
- pygments: 2.0.2

**Comment by ColinDuquesnoy** [comment]
Fixed by 72cebef9e31525c43760d71e2a18ce967b85f769

**Comment by ColinDuquesnoy** [comment]
I cannot reproduce this issue, the following message pop up (instead of your crash):

```
`"Cette version de C:/Program Files (x86)/OpenCobolIDE/GnuCOBOL/bin\\gcc.exe n'est pas compatible avec la version de Windows actuellement ex‚cut‚e. V‚rifiez dans les informations systŠme de votre ordinateur si une version x86 (32ÿbits) ou x64 (64ÿbits) du programme est requise, puis contactez l'‚diteur de logiciel.\r\n"
```

But I think this won't happen anymore, if there is a decoding error, the failure will get logged and will appear in the issues tab (the encoding that has been used to decode the bytes output should also happer).

Fixed by 72cebef9e31525c43760d71e2a18ce967b85f769
